servicetalk-http-router-jaxrs/src/test/java/io/servicetalk/http/router/jaxrs/HttpJaxRsRouterBuilderTest.java

```diff
@@ -0,0 +1,410 @@
+            return buildStringResponse("b", responseFactory);
+        }
+
+        @Path("/b/{c}")
+        Single<StreamingHttpResponse> param(@PathParam("c") String c,
+                                            @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(c, responseFactory);
+        }
+
+        @Path("/c/{a}/{b}")
+        Single<StreamingHttpResponse> paramAb(@PathParam("a") String a, @PathParam("b") String b,
+                                              @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(a + b, responseFactory);
+        }
+
+        @Path("/h/{a}/{b}")
+        Single<StreamingHttpResponse> paramAb(@PathParam("a") int a, @PathParam("b") int b,
+                                              @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse("" + a + b, responseFactory);
+        }
+
+        @Path("/d")
+        Single<StreamingHttpResponse> header(@HeaderParam("a") CharSequence a,
+                                             @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(a.toString(), responseFactory);
+        }
+
+        @Path("/cookie")
+        Single<StreamingHttpResponse> cookie(@CookieParam("a") Cookie cookie,
+                                             @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(cookie.getValue(), responseFactory);
+        }
+
+        @Path("/query")
+        Single<StreamingHttpResponse> query(@QueryParam("a") String a,
+                                            @QueryParam("b") int b,
+                                            @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(a + b, responseFactory);
+        }
+
+        @Path("/queryDefault")
+        Single<StreamingHttpResponse> queryDefault(@QueryParam("a") @DefaultValue("0") String a,
+                                                   @QueryParam("b") @DefaultValue("1") int b,
+                                                   @Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse(a + b, responseFactory);
+        }
+
+        @Path("/text")
+        Single<StreamingHttpResponse> text(@Context StreamingHttpResponseFactory responseFactory) {
+            return buildStringResponse("1", responseFactory);
+        }
+
+        @POST
+        @Path("/e")
+        Single<StreamingHttpResponse> body(@Context StreamingHttpResponseFactory responseFactory,
+                                           Publisher<Model> modelPublisher) {
+            return succeeded(responseFactory.ok().payloadBody(modelPublisher,
+                    jsonSerializer.serializerFor(Model.class)));
+        }
+
+        @PUT
+        @Consumes(value = {TEXT_PLAIN})
```

Comment:
Where is this (aka content-type based routing) tested?

Reply:
Here - HttpJaxRsRouterBuilderTest.testHeaderContentType